src/dune_init.ml

```diff
@@ -172,29 +172,23 @@ end
 (** The context in which the initialization is executed *)
 module Init_context = struct
   type t =
-    { root : Path.t
-    ; dir : Path.t
+    { dir : Path.t
     ; project : Dune_project.t
     }
 
-  let make root_str path =
-    let root = Path.of_string root_str in
+  let make path =
     let project =
-      match Dune_project.load ~dir:root ~files:String.Set.empty with
+      match Dune_project.load ~dir:Path.root ~files:String.Set.empty with
       | Some p -> p
       | None   -> Lazy.force Dune_project.anonymous
     in
     let dir =
       match path with
-      | None -> root
-      | Some path ->
-        if Filename.is_relative path then
```

Comment:
Without this, are users still able to specify either an absolute or a relative path as the location for the added component?

Reply:
I think I forgot to include a test case for that behavior. I'll try to add that tonight if I get time.

Reply:
Path.of_string should be smart enough to interpret relative paths correctly. A test would be useful indeed.

Reply:
Path.of_string should be smart enough to interpret relative paths correctly. A test would be useful indeed.